mcts.py

```diff
@@ -58,9 +68,17 @@ def search(self, initialState):
                 self.executeRound()
 
         bestChild = self.getBestChild(self.root, 0)
-        return self.getAction(self.root, bestChild)
+        action,node=((action, node) for action, node in self.root.children.items() if node is bestChild).__next__()
+        if needNodeValue:
+            nodeValue = node.totalReward / node.numVisits
+            return action, nodeValue
```

Comment:
Rather than just returning a tuple, can we return a dict of properties about the action so we can more easily extend this method in future with other properties as required, e.g. {"action": action, "expectedReward": node.totalReward / node.numVisits}? 

I'd also rather return this in all situations rather than have the return type change based on a flag, though that would be a backwards incompatible change so would require a major version bump. On balance, I'm happy for this change to go out behind a flag like this (though could we also rename this to reflect the above change), then in the next major version I'll make this the default behaviour and drop the flag.